app/services/katello/pulp3/repository.rb

```diff
@@ -166,10 +166,10 @@ def compute_remote_options(computed_options = remote_options)
         computed_options.except(:name, :client_key)
       end
 
-      def create
-        unless repository_reference
+      def create(force = false)
+        if force || !repository_reference
           response = api.repositories_api.create(create_options)
-          RepositoryReference.create!(
+          RepositoryReference.first_or_create!.update(
```

Comment:
FIXME: This is failing when the repo reference doesn't already exist because there are no arguments to create the first one.